src/components/license-activation/data/service.js

```diff
@@ -1,7 +1,17 @@
-import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import {
+  getAuthenticatedHttpClient,
+  fetchAuthenticatedUser,
+  hydrateAuthenticatedUser
+} from '@edx/frontend-platform/auth';
 import qs from 'query-string';
 
-export function activateLicense(activationKey) {
+export async function activateLicense(activationKey) {
+  // If the user has not refreshed their JWT since they created their account,
+  // we should refresh it so that they'll have appropriate roles (if available),
+  // and thus, have any appropriate permissions when making downstream requests.
+  const loginRefreshUrl = `${process.env.LMS_BASE_URL}/login_refresh`;
+  const refreshResult = await getAuthenticatedHttpClient().post(loginRefreshUrl);
```

Comment:
If we're not using `refreshResult`, I think we can just do:

```
await getAuthenticatedHttpClient().post(loginRefreshUrl);
```

Also, I'm wondering if we'll want to have this in a try/catch and potentially have similar logic as here: https://github.com/edx/frontend-platform/blob/db97454588dae1af718321b30908279e36bea191/src/auth/AxiosJwtTokenService.js#L54 (lines 65-77).

If the request fails, and it's because the user is not authenticated (401), remove the cookie for this situation: "to eliminate any situation where the cookie is not expired but the jwt is expired."?

Reply:
I like it in theory, but then this MFE has to know about the JWT cookie name, which is outside it purview (I think).

Reply:
...which I guess you could also say about the name of the `/login_refresh` route...

Reply:
All MFEs should already be aware of JWT cookie name: https://github.com/edx/edx-internal/blob/master/frontends/common/prod_config.yml#L11

Reply:
Accessible via `process.env.ACCESS_TOKEN_COOKIE_NAME` for example.

Reply:
hmm, ok, I'll give it a shot.